traits/traits.py

```diff
@@ -345,16 +358,30 @@ def __call__(self, *args, **metadata):
     @property
     def default(self):
         kind, value = self.default_value()
-        if kind in (2, 7, 8):
+        if kind in (
+            OBJECT_DEFAULT_VALUE,
+            CALLABLE_AND_ARGS_DEFAULT_VALUE,
+            CALLABLE_DEFAULT_VALUE,
+        ):
             return Undefined
-
-        if kind in (4, 6):
+        elif kind in (
+            DICT_COPY_DEFAULT_VALUE,
+            TRAIT_DICT_OBJECT_DEFAULT_VALUE,
+            TRAIT_SET_OBJECT_DEFAULT_VALUE,
+        ):
             return value.copy()
-
-        if kind in (3, 5):
+        elif kind in (
+            LIST_COPY_DEFAULT_VALUE,
+            TRAIT_LIST_OBJECT_DEFAULT_VALUE,
+        ):
             return value[:]
```

Comment:
It was very tempting to drop Python 2 support here and just use `value.copy()` for all the dict/list/set types.

Hmm. Actually, that makes me wonder whether `copy` does the right thing on `TraitListObject`.

Reply:
> that makes me wonder whether `copy` does the right thing on `TraitListObject`

It does, provided that "the right thing" means returning a plain list rather than a new `TraitListObject`.